spec/customer_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'customer'
+
+describe Customer do
+
+subject(:customer) { described_class.new(takeaway_klass) }
+let(:customer_klass) { double(new: customer, view_menu: takeaway.menu)}
+let(:takeaway_klass) { double(new: takeaway) }
+let(:takeaway) { double(menu: {chips: 4.00, rice: 2.00 }) }
+
+    describe '#view_menu' do
+      it 'shows a customer a menu from a specified takeaway' do
+        expect(customer_klass.view_menu(takeaway)).to eq(takeaway.menu)
+      end
+    end
+
+    describe '#initialize' do
+      it 'lets a customer select a takeaway' do
+        expect(customer.takeaway).to eq(takeaway)
+      end
+    end
+
+    describe '#select_dish' do
+      it 'adds dishes to the customer\'s basket' do
+        customer.select_dish(:chips, 2)
+        expect(customer.basket).to eq({chips: 2})
+      end
+
+      it 'raises an error if the selected dish does not exist' do
+        error = 'Please select a dish from the menu'
+        expect{customer.select_dish(:invalid_dish, 1)}.to raise_error error
+      end
+    end
+
+    describe '#order_total' do
+      it 'provides the customer with the total of all individual items in their basket' do
```

Comment:
Metrics/LineLength: Line is too long. [90/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)